server/serverAPI/controller/AuthenticationController.ts

```diff
@@ -101,6 +118,72 @@ export default class AuthenticationController extends BaseUserController {
         }
     }
 
+    confirmVerificationCode = async (req: Request, res: Response) => {
+        let username = req.body?.username;
+        let actualCode = req.body?.verificationCode;
+
+        let requestedCode = AuthenticationController.verificationCodesMap.get(username);
+
+        if (requestedCode === undefined) {
+            return this.sendError(400, res, "Verification code is either expired or not issued.");
+        }
```

Comment:
This also implicitly handles the case where the `username` doesn't exist in the database. Can you provide an error for that either explicitly (look up the username and specify if the user doesn't exist) or implicitly (update this error message to say the username may not be registered)? I prefer the explicit approach